### ISSUE TITLE
Add the ability to collaborate with the same code

### ISSUE THREAD

**Issue by Dragon20C** [godotengine/godot-proposals#2398]
### Describe the project you are working on

The ability to live share the same code and see each other change bits of the code, here and there.

### Describe the problem or limitation you are having in your project

The ability to work as a team efficiently 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Having this feature would mean there would be less time downloading each others project/code and less time on refactoring each others code, to work together.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

I am not sure, But I would assume it would use a type of syncing system that detects when something is changed and send an update to the users or it could be a type of streaming where users can see changes made by other users.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

That would be just downloading the project and then changing the code that has changed which could be time consuming

### Is there a reason why this should be core and not an add-on in the asset library?

None, this could be an add-on but I think this being made inside of godot would be a good way of attracting user base that work in teams.

**Comment by pycbouh** [comment]
Are you familiar with version control systems, such as Git?

**Comment by Dragon20C** [comment]
> Are you familiar with version control systems, such as Git?

I am not, does it produce similer results in that the user/developer doesnt need to download or refactor code?
I have used github though limmited amounts.

**Comment by Zireael07** [comment]
Dragon20C, yes, a Git user needs to download code (git pull) but since it only tracks changes, it means less data to download, which means faster to update.

Also I think the proposal is out of scope for Godot itself. Godot is already plenty friendly for version control such as git, having textual files instead of binary.

**Comment by pycbouh** [comment]
> I am not, does it produce similer results in that the user/developer doesnt need to download or refactor code?

You cannot avoid "downloading" code if you want to share it with someone else. There is no magic, you need to get changes made by someone else on their PC to your PC. Version control systems provide you with a tool to do so in an organized manner. You have a repository (such as this one we are currently in), that is copied among a remote server and all your developers, and each developer does their changes locally, then pushes them to the remote server from which you can get it. The process, once you get used to it, is straighforward and transparent.

I'm not sure what kind of refactoring you refer to, as refactoring is a process of rewriting a part of a code base in a new way, possible more organized or more optimized. When collaborating you don't usually refactor other's code, you just pull it, apply it to your local copy of the project. VCS help here by allowing code can be contained within different branches, as well as providing tools for intelligently merging different patches of code (commits) with a lot of automation.

Long story short, if you want to collaborate, you should learn how to use version control systems, Git being the most widespread one (but other options exist).

**Comment by Dragon20C** [comment]
@pycbouh I am trying to keep it realistic maybe I shouldnt mention project more code specific the reason I mention something like this is because I found something to do with vscode, where you can live share / collaborate with each other on code

**Comment by Dragon20C** [comment]
sorry guys and gals if this isn't a good idea please close it, if it gets a negative or confused response

**Comment by Xrayez** [comment]
> Also I think the proposal is out of scope for Godot itself. Godot is already plenty friendly for version control such as git, having textual files instead of binary.

See https://github.com/godotengine/godot-git-plugin.

@Dragon20C did you mean #851 perhaps?

**Comment by Jummit** [comment]
> None, this could be an add-on but I think this being made inside of godot would be a good way of attracting user base that work in teams.

Shouldn't that rule out this proposal entirely? Or am I missing something about the GIP workflow?

**Comment by pycbouh** [comment]
> Shouldn't that rule out this proposal entirely (Rule 3.)? Or am I missing something about the GIP workflow?

I'm keeping it open in case this is not what it looks like at the first glance.

**Comment by Dragon20C** [comment]
@Xrayez exactly that, the ability to have someone come in and edit the code as they wish, what would be the correct term , live sharing?

**Comment by Zireael07** [comment]
In that case, vote to close as a duplicate of #851.

**Comment by Dragon20C** [comment]
Closed because of dup
